projects/storefrontlib/src/events/product/product-page-event.builder.ts

```diff
@@ -72,22 +72,16 @@ export class ProductPageEventBuilder {
       skip(1)
     );
 
-    const categoryPageEvent$ = this.eventService.get(PageEvent).pipe(
-      map((pageEvent) => ({
-        isCategoryPage: pageEvent.semanticRoute === 'category',
-        categoryCode: pageEvent.context.id,
-      }))
-    );
-
-    return categoryPageEvent$.pipe(
+    return this.eventService.get(PageEvent).pipe(
       switchMap((pageEvent) => {
-        if (!pageEvent.isCategoryPage) {
+        if (pageEvent.semanticRoute !== 'category') {
           return EMPTY;
         }
 
         return searchResults$.pipe(
           map((searchResults) => ({
-            categoryCode: pageEvent.categoryCode,
+            ...pageEvent,
+            categoryCode: pageEvent.context.id,
             categoryName: searchResults.breadcrumbs[0].facetValueName,
```

Comment:
can we drop the potential null pointer here? I've already had reports on that, not necessarily in this code.

Reply:
Good catch, done